### ISSUE TITLE
Allow subproperties for queries (via message field) and subsets of properties in message fields via ToMessage

### ISSUE THREAD

**Issue by dhermes** [GoogleCloudPlatform/endpoints-proto-datastore#53]
Support `attr.subattr` for `StructuredProperty`s, likely a change in `MessageFieldsSchema`, but possibly elsewhere? This would also solve the problem of passing property specific orderings down, but would have the adverse affect of creating possible name collisions and unnecessary re-definitions of ProtoRPC message classes, if done incorrectly.

This is actually done in NDB in projections for `StructuredProperty`s.

Not supporting this is related to the use of `allow_message_fields` in `ProtoModel`.

For the three original TODO comments see:
- https://github.com/GoogleCloudPlatform/endpoints-proto-datastore/blob/696638e50dab9a0f4010aed3b6678a480859a1aa/endpoints_proto_datastore/ndb/model.py#L86
- https://github.com/GoogleCloudPlatform/endpoints-proto-datastore/blob/696638e50dab9a0f4010aed3b6678a480859a1aa/endpoints_proto_datastore/ndb/model.py#L1368
- https://github.com/GoogleCloudPlatform/endpoints-proto-datastore/blob/696638e50dab9a0f4010aed3b6678a480859a1aa/endpoints_proto_datastore/ndb/model.py#L1447

<sub>**NOTE:** THIS POST WAS EDITED AFTER BEING MIGRATED.</sub>

---

---

<sub>**Code Hosting Comment Metadata:**</sub>
<sub>author.htmlLink=https://code.google.com/u/dhermes@google.com/</sub>
<sub>author.name=dhermes@google.com</sub>
<sub>id=0</sub>
<sub>published=2012-11-29T07:36:07.000Z</sub>

<sub>**Code Hosting Issue Metadata:**</sub>
<sub>published=2012-11-29T07:36:07.000Z</sub>
<sub>stars=1</sub>
<sub>updated=2013-01-18T06:46:01.000Z</sub>
<sub>originalIssue=https://code.google.com/p/endpoints-proto-datastore/issues/detail?id=7</sub>
<sub>author=@dhermes</sub>

**Comment by matantsu** [comment]
how do I use it inside `query_method` , or is it not implemented yet ?

**Comment by matantsu** [comment]
how do I use it inside `query_method` , or is it not implemented yet ?

**Comment by dhermes** [comment]
Do you really need this? It's quite a mess to offer something like

```
/query/path?arg1=foo&prop.subprop1=bar&prop.subprop2=baz
```

when the pragmatic choice

```
/query/path?arg1=foo&subprop1=bar&subprop2=baz
```

is readily available.

**Comment by matantsu** [comment]
well , I came across this trying to implement an api for searching entities by `Address` , and an `Address` is an entity containing things like `zip_code , street , city , etc...`. so every searchable entity will have a `ndb.StructuredProperty(Address)`

so I thought it was obvoius to put `query_fields=('Address',)`

**Comment by dhermes** [comment]
I suppose it is obvious, but this is a non-trivial feature to implement.

As a stop-gap you can just implement custom `EndpointsAliasProperty`s to populate the query in the address.

I think you'll find (see the [App Engine way][1]) that you'd rather just store the properties you're querying for on the object as properties than as properties of a property (`Address`).

[1]: http://bjk5.com/post/54202245691/the-app-engine-way

**Comment by matantsu** [comment]
yes , I have looked at that , but for me code quality and readibility is more important than performance.
I would rather split my data into logical units (objects) and suffer the penalty of performance than simply throwing it all in.
